lib/fluent/plugin/filter_kubernetes_metadata.rb

```diff
@@ -253,13 +269,39 @@ def log.trace?
 
     end
 
+    def shutdown
+      super
+      if @checkpoint_enabled
+        write_cache_to_file
+        @checkpoint_thread_mutex.synchronize do
```

Comment:
Fix this use of mutex too as discussed; In line with the comment above of the mutex and variable actually being used to do this